portfolio/src/main/webapp/script.js

```diff
@@ -452,6 +458,78 @@ function logoutHandler(value){
     //removeClass("chart-container", "blur-content"); not necessary for this pr
 }
 
+//*********************** VOTING HANDLING ***********************
+/**
+ * Listens for the selected genre by the user
+ */ 
+function listenForGenreChoice(){
+    const form = document.getElementById("chartForm");
+    const el = document.getElementsByName('genre'); 
+    const sendingParam = document.getElementById("selectedRadio");
+    let selectedValue = null;
+    
+    for(let i = 0; i < el.length; i++) { 
+        if(el[i].checked){
+            selectedValue = el[i].value;
+        } 
+    }
+
+    form.onsubmit = function(e){
+
+        //if not logged in
+        if (!currentUser.active) {
+            //normally this affects blur, will see in style pr
+            alert("you need to log in to use this");
+            e.preventDefault();
+            return;
+        }
+
+        //if the user has already posted
+        if(currentChartData.vUsers.includes(currentUser.id)) {
+            alert("you've already put in your 2wo cents");
+            e.preventDefault();
+            return;
+        }
+
+        // get selected value          
+        for(let i = 0; i < el.length; i++) { 
+            if(el[i].checked){
+            selectedValue = el[i].value;
+            } 
+        }
+
+        if (selectedValue !== null) {
+            setParameters(selectedValue);
+            document.chartForm.submit();
+        } else {
+            alert("you need to select a radio value to submit");
+            e.preventDefault();
+        }
+    }
+}
+
+/**
+ * Retrieve genre from server
+ */ 
+async function getGenreChoice(){
+    const response = await fetch('/chart');
+    const value = await response.json();
+    currentChartData = value;
```

Comment:
do you display this chart data anywhere?

Reply:
Yes this gets displayed on click of show chart button